0002/README.md

```diff
@@ -0,0 +1,232 @@
+    reach SLSA 4 for specific languages (go, Python, etc.) These will hopefully
+    prove ideas and provide more data, but we do not expect to support these
+    long term. ([@laurentsimon], P1)
+*   Proposal for how to generalize and scale ecosystem-specific solutions to
+    achieve SLSA 4 and generate high quality SBOMs that can be maintained by
+    each community independently, without risk of accidentally breaking SLSA
+    guarantees. ([@laurentsimon], P2)
+
+#### Milestone: Provenance distribution model is decided, with at least one representative implementation (tentatively Python)
+
+Consumers have a standard mechanism for finding and retrieving provenance of the
+packages they use within each packaging ecosystem. The tentative plan is to
+distribute provenance in-band next to the artifact, plus have an out-of-band
+mechanism for prototyping and supporting attestations from parties other than
+the producer. This work also includes aligning with SBOM, since the requirements
+are very similar. With this milestone, motivated consumers can act upon the
+provenance themselves.
+
+Design is P1 for [@lumjjb] in Q2.
+
+Issue: [#269](https://github.com/slsa-framework/slsa/issues/269)
+
+#### Milestone: SLSA policy model is decided, to detect or prevent publishing of packages containing unauthorized changes, with at least one representative implementation (tentatively Python)
+
+SLSA policies detect or prevent unauthorized changes for packages within the
+ecosystem. When opted-in, package maintainers must upload provenance alongside
+the package proving that the package was built directly from the canonical
+source repository. At first this may just display results on a dashboard,
+similar to [pyreadiness.org](https://pyreadiness.org/3.10/), but will hopefully
+be integrated directly into PyPI eventually. With this milestone, software
+producers can opt-in to automatically protect themselves against insider risk,
+and software consumers can gain assurance for opted-in packages, but the package
+registry is still trusted.
+
+Design is P1 for [@var-sdk] and [@wietse-gmail] in Q2.
+
+Sub-problems:
+
+-   Policy model, which securely maps package name to canonical source
+    repository. For example, [requests](https://pypi.org/project/requests/) maps
+    to https://github.com/psf/requests. This metadata already exists for some
+    packages, but (a) it is not structured and (b) package maintainers can
+    mutate it at will. The solution must protect against malicious changes while
+    silently allowing official changes in source repository. One idea is to have
+    a Trust On First Use (TOFU) model with a mechanism for source repo
+    migrations.
+
+-   Implementation for each specific ecosystem, starting with Python/PyPI. This
+    includes a design, prototype, dashboard, and/or upload-time integration.
+
+-   (optional) Toy implementation of a mock package registry ("SLSA Playground")
```

Comment:
A colleague at VMware (👋 @jku) is starting a mock package registry project – https://github.com/jku/repository-playground – with a goal of providing a place to develop ideas related to the OpenSSF [Securing Software Repos WG](https://github.com/ossf/wg-securing-software-repos).

It would be fantastic if we could share mock package registry implementation to enable folks to focus more on the security problems they are trying to solve.

Reply:
Awesome! Great minds think alike! 😉  I'd love to use this to show how SLSA works! Added a link to the doc.